language-api/src/main/java/de/jplag/util/FileUtils.java

```diff
@@ -0,0 +1,141 @@
+package de.jplag.util;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import com.ibm.icu.text.CharsetDetector;
+import com.ibm.icu.text.CharsetMatch;
+
+/**
+ * Encapsulates various interactions with files to prevent issues with file encodings.
+ */
+public class FileUtils {
+    private static final Charset defaultOutputCharset = StandardCharsets.UTF_8;
```

Comment:
Constants should be named like this ..
```suggestion
    private static final Charset DEFAULT_OUTPUT_CHARSET = StandardCharsets.UTF_8;
```